app/auth/auth.py

```diff
@@ -0,0 +1,59 @@
+from flask import g, jsonify, request
+from app import app
+from flask_httpauth import HTTPBasicAuth
+from flask_login import login_required, login_user, logout_user
+from app.models import db, User
+
+auth = HTTPBasicAuth()
+
+
+@auth.verify_password
+def verify_token(token, password):
+    # authenticate by token only
+    user = User.verify_auth_token(token)
+    if not user:
+        return False
+    g.user = user
+    return True
+
+
+@app.route('/auth/register', methods=['POST'])
+def register():
+    """Register a user"""
+    email = request.json.get('email')
+    password = request.json.get('password')
+    if email is None or password is None:
+        return jsonify(message='missing arguments!'), 400
```

Comment:
Nice catch! You can use a more descriptive error message for a layman though. `Email or password is missing`